packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -229,6 +232,10 @@ class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
   String get apksignerPath => getBuildToolsPath('apksigner');
 
   List<String> validateSdkWellFormed({bool requireApkSigner = true}) {
+    if (buildToolsVersion.major < minimumAndroidSdkVersion) {
```

Comment:
Makes sense for build and platform tools to be in sync. But there are AFAIK actually four artefacts in total: SDK Platform, SDK Build Tools, SDK Platform-tools, and SDK Tools:
https://developer.android.com/studio/intro/update.html#sdk-manager

Must they all four be in sync?

I thought not -- for example, prior to https://github.com/flutter/flutter/pull/8461/files our samples were using SDK 25 but build tools 24. 

Or am I totally misunderstanding this?